projects/fsastorefrontlib/src/cms-components/checkout/components/cart-coupon/cart-coupon.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import {
+  ActiveCartService,
+  CartVoucherService,
+  CustomerCouponService,
+  OCC_USER_ID_CURRENT,
+  UserIdService,
+} from '@spartacus/core';
+import { CartCouponComponent } from '@spartacus/storefront';
+import { Observable } from 'rxjs';
+import { FSCartService } from '../../../../core/cart/facade/cart.service';
+
+@Component({
+  selector: 'cx-fs-cart-coupon',
+  templateUrl: './cart-coupon.component.html',
+})
+export class FSCartCouponComponent extends CartCouponComponent
+  implements OnInit, OnDestroy {
+  currentUser;
+  userRegistered$: Observable<any>;
+
+  constructor(
+    protected cartVoucherService: CartVoucherService,
+    protected formBuilder: FormBuilder,
+    protected customerCouponService: CustomerCouponService,
+    protected activeCartService: ActiveCartService,
+    protected cartService: FSCartService,
+    protected userIdService: UserIdService
+  ) {
+    super(cartVoucherService, formBuilder, customerCouponService, cartService);
+    this.currentUser = OCC_USER_ID_CURRENT;
```

Comment:
Constructor is not needed here 🤔 
Initialization of current user can be moved in ngOnInit

Reply:
It's needed, because we instantiate `UserIdService`.
And initialization of current user was originally in ngOnint, but I moved it to constructor. I thought it would be safer to have it initialize before everything else. Don't see any harm in this approach...

Reply:
On second thought, this variable is not needed at all. We can just check on the template if userRegistered is equal to 'current'. Now, we are assigning value to a variable using another readonly constant.

Reply:
Yup, firstly I did it like that, but thought that it is maybe ugly to have it fixed like that, and more importantly maybe Spartacus changes it's implementation for `current` user to something else in the future. What do you think? Should I revert this anyway?

Reply:
I would revert it. 'current' is value used on backend so it cant be changed

Reply:
Okay, then, it shall be so...